tests/e2e/instrumentation-golang/01-install-app.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment-with-sidecar
+spec:
+  selector:
+    matchLabels:
+      app: my-pod-with-sidecar
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: my-pod-with-sidecar
+      annotations:
+        sidecar.opentelemetry.io/inject: "true"
+        instrumentation.opentelemetry.io/inject-golang: "true"
+        instrumentation.opentelemetry.io/golang-target-exec: /usr/local/bin/emojivoto-voting-svc
+    spec:
+      containers:
+      - name: voting-svc
+        image: docker.l5d.io/buoyantio/emojivoto-voting-svc:v11
```

Comment:
how was this image chosen? is the license for the image permissive of us using it?

Reply:
It was chosen based on the existing opentelemetry-go-instrumentaion docs example, but the repo has some new e2e tests with their own sample app that we could probably use now instead.

Reply:
The opentelemetry-go-instrumentation repo's sample app is build during the e2e workflow and isn't currently published.  The license for https://github.com/BuoyantIO/emojivoto is [Apache License 2.0](https://github.com/BuoyantIO/emojivoto/blob/main/LICENSE) so believe we are allowed to use it.  I will add a link back to the repo like the dotnet image does.

Reply:
Actually, let me try one of the opentelemetry-demo Go images.

Reply:
If it helps we could probably publish an image for one of the sample apps

Reply:
This particular e2e tests only tests that the operator can add the sidecar, not that the sidecar produces meaningful telemetry.  I feel the demo image is sufficient and saves everyone extra work.